rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -2183,26 +2181,49 @@ void rskTxWaitingForSignature_fail_adding_an_already_existence_key_after_rskip_3
         bridgeSupport.updateCollections(tx);
 
 
-        // Assert two pegouts are added to pegouts waitingForConfirmations, one pegout batch and one pegout migration tx
-        assertEquals(2, provider.getPegoutsWaitingForConfirmations().getEntries().size());
-        // Assert the two added pegouts has the same pegoutCreationRskTxHash
-        assertEquals(1,
-            provider
-                .getPegoutsWaitingForConfirmations()
-                .getEntries()
-                .stream()
-                .map(PegoutsWaitingForConfirmations.Entry::getPegoutCreationRskTxHash)
-                .collect(Collectors.toSet()).size()
+        // Assert two transactions are added to pegoutsWaitingForConfirmations, one pegout batch and one migration tx
+        assertEquals(2, pegoutsWaitingForConfirmations.getEntries().size());
+
+        // Get new fed wallet to identify the migration tx
+        Wallet newFedWallet = BridgeUtils.getFederationNoSpendWallet(
+            new Context(bridgeConstants.getBtcParams()),
+            newFederation,
+            false,
+            null
         );
 
+        PegoutsWaitingForConfirmations.Entry migrationTx = null;
+        PegoutsWaitingForConfirmations.Entry pegoutBatchTx = null;
+
+        /*
+            Iterate pegoutsWaitingForConfigrmation entries to identify migration tx by checking if all outputs are sent to the active fed means this is
+            a migration tx; if not, it is a peg-out batch
+         */
+        for (PegoutsWaitingForConfirmations.Entry entry : pegoutsWaitingForConfirmations.getEntries()) {
+            List<TransactionOutput> walletOutputs = entry.getBtcTransaction().getWalletOutputs(newFedWallet);
+            if (walletOutputs.size() == entry.getBtcTransaction().getOutputs().size()){
+                migrationTx = entry;
+                continue;
+            } else {
+                pegoutBatchTx = entry;
+            }
+        }
```

Comment:
This was smart. I have another idea though, since we already asserted above that there are 2 transactions waiting for confirmation we could also take the chance to assert that the migration transaction is the one created first. So this could like 

```
PegoutsWaitingForConfirmations.Entry migrationTxEntry = pegoutsWaitingForConfirmations.getEntries().get(0);
PegoutsWaitingForConfirmations.Entry pegoutBatchTxEntry = pegoutsWaitingForConfirmations.getEntries().get(1);

List<TransactionOutput> migrationTxOutputs = migrationTxEntry.getBtcTransaction().getWalletOutputs(newFedWallet);
assertEquals(migrationTxOutputs, migrationTxEntry.getBtcTransaction().getOutputs().size());
```
And add a comment above explaining why we are doing that assertion. What do you think?

Reply:
Nice. I like it. Since the first method call by `updateCollections` is `processFundsMigration` the migration tx should be the first to be added to the `pegoutsWaitingForConfirmations`.